### PR TITLE
feat(string): add padCharsEnd

### LINKED ISSUE #1104: padCharsEnd
**Is your feature request related to a problem? Please describe.**

This function pads the current string with a given string (repeated, if needed) so that the resulting string reaches a given length. The padding is applied from the end of the current string.

**Describe the solution you'd like**

```js
RA.padCharsEnd('-', 3, 'a'); // => 'a--'
```

The function is of arity=3 and is autocurried.

**Additional context**

The functions should use [String.prototype.padEnd()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd) if available. If not polyfill like [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd#Polyfill) should be provided to compensate.

[This PR](https://github.com/char0n/ramda-adjunct/pull/1067) should serve as a model template of PR for `padCharsEnd` function.

### PR DESCRIPTION
**Pull request by dshubhadeep**
Closes #1104

### CODE REVIEW

File: src/internal/polyfills/String.padCharsEnd.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const padCharsEnd = (padString, targetLength, value) => {
```
Comment (char0n) [inline src/internal/polyfills/String.padCharsEnd.js]: padCharsEnd -> padCharsEndPolyfill

File: src/internal/polyfills/String.padCharsEnd.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  let finalPadString = String(
+    typeof padString !== 'undefined' ? padString : ' '
+  );
+
+  if (value.length > finalLength) {
+    return String(value);
+  }
+  finalLength -= value.length;
+  if (finalLength > finalPadString.length) {
+    finalPadString += finalPadString.repeat(
```
Comment (char0n) [inline src/internal/polyfills/String.padCharsEnd.js]: String.repeat may not be available, so we have two options here:

**1.** use always polyfill of [String.prototype.repeat](https://github.com/char0n/ramda-adjunct/blob/f1c5dfaac3b38a5599a6bb71a570205b025c7112/src/internal/polyfills/String.repeat.js#L50-L49) here that we already have in codebase

**2.** use polyfill of [String.prototype.repeat](https://github.com/char0n/ramda-adjunct/blob/f1c5dfaac3b38a5599a6bb71a570205b025c7112/src/internal/polyfills/String.repeat.js#L50-L49) only when `repeat` method on string is not available. 

I'll leave the decission and implementation up to you. IMHO best approach is second option

File: src/padCharsEnd.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export const padCharsEndStrInvoker = flip(invoker(2, 'padEnd'));
+
+/**
+ * The function pads the current string with a given string
+ * (repeated, if needed) so that the resulting string reaches a given length.
+ * The padding is applied from the end of the current string.
+ *
+ * @func padCharsEnd
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.11.0|v2.11.0}
```
Comment (char0n) [inline src/padCharsEnd.js]:
```js
@since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
```

File: test/padCharsEnd.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  padCharsEndStrPolyfill,
+} from '../src/padCharsEnd';
+import eq from './shared/eq';
+
+describe('padCharsEnd', function() {
+  specify('should pad string with given padString', function() {
+    eq(RA.padCharsEnd('-', 3, 'a'), 'a--');
+  });
+
+  specify('should be curried', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: Usually we cant currying tests as last tests in the block. This is just a convention that we use.

File: test/padCharsEnd.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+      if (RA.isNotFunction(String.prototype.padEnd)) {
+        this.skip();
+      }
+    });
+
+    specify('should pad string with given padString', function() {
+      eq(padCharsEndStrInvoker('-', 3, 'a'), 'a--');
+    });
+
+    specify('should be curried', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: Usually we cant currying tests as last tests in the block. This is just a convention that we use.

File: test/padCharsEnd.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  context('given targetLength less than equal to padString length', function() {
+    specify('should trim padString', function() {
+      eq(RA.padCharsEnd('123456', 6, 'abc'), 'abc123');
+      eq(RA.padCharsEnd('123456', 5, 'abc'), 'abc12');
+    });
+  });
+
+  context('given targetLength supplied as float', function() {
+    specify('should convert targetLength to integer', function() {
+      eq(RA.padCharsEnd('$', 7, 'abc'), 'abc$$$$');
```
Comment (char0n) [inline test/padCharsEnd.js]: We're passing an integer here instead of float

File: test/padCharsEnd.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    });
+  });
+
+  context('given targetLength is not a number', function() {
+    specify('should return padded string', function() {
+      eq(RA.padCharsEnd('*', '5', 'abc'), 'abc**');
```
Comment (char0n) [inline test/padCharsEnd.js]: This look more like `given targetLength is a value coercible to valid integer number`

File: test/padCharsEnd.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+    });
+  });
+
+  context('given empty padString', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: `given padString set to undefined`

File: test/padCharsEnd.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+      }
+    );
+
+    context('given targetLength supplied as float', function() {
+      specify('should convert targetLength to integer', function() {
+        eq(padCharsEndStrInvoker('$', 7, 'abc'), 'abc$$$$');
+      });
+    });
+
+    context('given targetLength is not a number', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: This look more like `given targetLength is a value coercible to valid integer number`

File: test/padCharsEnd.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+      specify('should return padded string', function() {
+        eq(padCharsEndStrInvoker('*', '5', 'abc'), 'abc**');
+      });
+    });
+
+    context('given empty padString', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: `given padString set to undefined`

File: test/padCharsEnd.js [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+      });
+    });
+  });
+
+  context('padCharsEndStrPolyfill', function() {
+    specify('should pad string with given padString', function() {
+      eq(padCharsEndStrPolyfill('-', 3, 'a'), 'a--');
+    });
+
+    specify('should be curried', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: Usually we cant currying tests as last tests in the block. This is just a convention that we use.

File: test/padCharsEnd.js [hunk 12]
```diff
@@ -0,0 +1,185 @@
+import * as RA from '../src';
+import {
+  padCharsEndStrInvoker,
+  padCharsEndStrPolyfill,
+} from '../src/padCharsEnd';
+import eq from './shared/eq';
+
+describe('padCharsEnd', function() {
```
Comment (char0n) [inline test/padCharsEnd.js]: Can we add tests that demonstrates what happens when `targetLength` is a negative number ?

File: types/index.d.ts [hunk 13]
```diff
@@ -1243,6 +1243,14 @@ declare namespace RamdaAdjunct {
          */
         allUnique<T>(list: T[]): boolean;
 
+        /**
```
Comment (char0n) [inline types/index.d.ts]: The description should exactly match the description provided in `padCharsEnd.js`.
```js
* The function pads the current string with a given string
 * (repeated, if needed) so that the resulting string reaches a given length.
 * The padding is applied from the end of the current string.
